core/query/query.ml

```diff
@@ -378,7 +390,8 @@ struct
       | `PrimitiveFunction (f,_) -> Primitive f
           (*     | `ClientFunction f ->  *)
```

Comment:
Can we delete these commented-out cases?

Reply:
Yep. I've also taken the opportunity to print the value, so we get a more informative error message.